changelog_unreleased/cli/15533.md

```diff
@@ -0,0 +1,5 @@
+#### Process files only supported by plugins (#15533 by @sanmai-NL)
+
+In #14627, Prettier was redesigned to ‘no longer follow symbolic links’. However, the implementation went further and turned into a breaking change. Rather than adhering to the exit status contract of only returning a non-zero exit status when a file is not formatted or when something is wrong with Prettier itself, #14627 made Prettier return a non-zero exit status upon any pattern turning out to be a pointing to symlink.
```

Comment:
I take exception to this mischaracterization of #14627. Prettier has always exited with code 2 for an unsupported file (e.g. `touch foo.xyz; prettier foo.xyz`). With #14627, symlinks are intentionally unsupported files, so it is appropriate to exit with code 2. The implementation adheres to the contract, and this behavior was specifically [noted and agreed upon](https://github.com/prettier/prettier/pull/14627#pullrequestreview-1364278508) with the maintainer.

You could easily describe what you’re changing here without implying that I made some kind of mistake.

Reply:
It's a pity you only bring forward your argument now the PR is finished, even as you were pinged in the OP. In this PR or indeed in reply to my comments on your earlier PR, you could've responded with factual discussion.

An unmatched file is simply a branch of the business logic, an has nothing to do with a fault state. If what you're claiming is correct, the docs aren't correct in not taking the unmatched files fault (parameter default) into account.

Reply:
Stop arguing about what's correct or wrong. I'll take PR this over.

Reply:
I [did bring forward this argument](https://github.com/prettier/prettier/pull/14627#issuecomment-1754003937). It was clear even before my comments that your PR isn’t finished, as it’s failing CI; in any case, receiving and responding to review feedback is a normal, healthy part of the PR process. If you think some part of my discussion is not factual, you are welcome to explain.

I’ll repeat that #14627 *did not introduce* the convention that runtime errors exit with code 2. It seems to me that you’re reading too much into the documentation’s [very few words](https://prettier.io/docs/en/cli#exit-codes) on exit codes that it doesn’t actually say. If we consider the (again, pre-existing) case of the specified file being missing, for whatever argument you might have that it’s not precisely described by some narrow interpretation of code 1 “Something wasn’t formatted properly” or code 2 “Something’s wrong with Prettier”, there’s a much clearer case that it’s not correctly described by code 0 “Everything formatted properly”. But if you’d like to improve the documentation, my suggestion would be to open a separate PR for that (I’ll defer to the maintainers if they have a different suggestion).

My main concern in this comment is that I believe it’s unfair to characterize #14627 in the way that you have.

Reply:
@andersk , you make a circular argument. You argue against my PR changing behavior from your PR, just because it conflicts with your PR (‘With https://github.com/prettier/prettier/pull/14627, symlinks are intentionally unsupported files’). That behavior change was breaking and disables a valid use-case, as I already argued in your PR thread. I also discussed and you conceded that a proper enhancement would be to process symlinks and separate invalid/unacceptable ones from acceptable ones.

> The implementation adheres to the contract (which never said “Prettier itself”—

I quote the docs about exit status 2:

> Something’s wrong with Prettier

Period. My description ‘Prettier _itself_’ adds no meaning,  it only stresses the exact meaning of this sentence. 

> this is one of many runtime errors that are completely reasonable to categorize there)

Okay, so what do you think about this scale:

1. Invalid memory access.
2. Division by zero.
3. Non-existent configuration parameter
4. All files specified are unmatched.
5. One file unmatched because it does not exist.
6. Some files are deemed unmatched because they're broken or unacceptable symlinks.
7. One file among matched ones, is unmatched just because it is a symlink.

You made anything from 6 on result in a fault (an error is something a human does). For case 4. extended to, at most, unsupported files, one could stretch the meaning of fault, since there would be no files to check left. Just as well we could let Prettier log that and exit normally. But 7 is by no means a fault. Especially not when it is a perfectly valid symlink.

Reply:
@fisker  I didn't see your comment before responding. I will rest my case, even though I think there are more claims left here by @andersk that I disagree with.